### PR TITLE
Migrate Lineage functionality test to ProgramMetadataTest

### PR DESCRIPTION
**Pull request by jayfjin**
PurchaseApp contains flows and streams and thus will not be migrated.

Move lineage and search testing functionality to ProgramMetadataTest. ProgramMetadataApp is simpler than PurchaseApp but the core testing functionality remains the same.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -33,6 +33,8 @@ To run integration tests against a remote CDAP instance, execute::
 
 Note that the CDAP instance against which the integration tests are run must have unrecoverable reset enabled.
 
+Also note that if the UI is running under the default host/port of "localhost:11011", then the desired host to connect to is "localhost:11015".
```
Comment (anwar6953) [inline README.rst]: I think you can omit this and just say that if `-DinstanceUri` is not specified, then it connects to the default host/port.

File: integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/ProgramMetadataTest.java [hunk 2]
```diff
@@ -15,31 +15,216 @@
...
+                    RunIds.fromString(runRecords.get(0).getPid()))
+            )), ImmutableSet.of());
+    waitFor(expected, () -> lineageClient.getLineage(INPUT_DATASET, startTime, endTime, null));
+    waitFor(expected, () -> lineageClient.getLineage(OUTPUT_DATASET, startTime, endTime, null));
+  }
+
+  @Test
+  public void testSearchUsingSystemMetadata() throws Exception {
+    deployApplication(ProgramMetadataApp.class);
+    // search artifacts
```
Comment (anwar6953) [inline integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/ProgramMetadataTest.java]: i think these comments aren't necessary - they seem redundant to the method names.

File: integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/ProgramMetadataTest.java [hunk 3]
```diff
@@ -15,31 +15,216 @@
...
+        ImmutableSet.of()),
+        lineageClient.getLineage(OUTPUT_DATASET, startTime, endTime, null));
+
     MapReduceManager mapReduceManager =
       applicationManager.getMapReduceManager(PROGRAM.getProgram());
     mapReduceManager.start();
     mapReduceManager.waitForRun(ProgramRunStatus.COMPLETED, 5, TimeUnit.MINUTES);
+
+    List<RunRecord> runRecords = getRunRecords(1, programClient, PROGRAM,
+        ProgramRunStatus.COMPLETED.name(), 0, endTime);
```
Comment (anwar6953) [inline integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/ProgramMetadataTest.java]: indentation seems inconsistent from the rest of the projects.
Can you make sure you're using these settings: https://wiki.cask.co/display/CE/Coding+Standard

File: README.rst (hunk 1 above)
Comment (jayfjin) [reply README.rst]: Done

File: integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/ProgramMetadataTest.java (hunk 2 above)
Comment (jayfjin) [reply integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/ProgramMetadataTest.java]: Done

File: integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/ProgramMetadataTest.java (hunk 3 above)
Comment (jayfjin) [reply integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/ProgramMetadataTest.java]: Done. Thanks for the review, I'll squash after everything's finalized

**Pull request merged**